tests/libs/util/socket_helper.h

```diff
@@ -99,22 +145,32 @@ typedef class _receiver_socket : public _base_socket
     ~_receiver_socket();
     void
     complete_async_receive(bool timeout_expected = false);
+    virtual void
+    complete_async_send(int timeout_in_ms) = 0;
     void
-    get_received_message(_Out_ uint32_t& message_size, _Outref_result_buffer_(message_size) char*& message);
+    complete_async_receive(int timeout_in_ms, bool timeout_expected = false);
+    // void
+    // get_received_message(_Out_ uint32_t& message_size, _Outref_result_buffer_(message_size) char*& message);
 
     virtual void
     post_async_receive() = 0;
     virtual void
+    send_async_response(_In_z_ const char* message) = 0;
```

Comment:
2 points:

1. Be consistent in method naming. Either stick to send/receive_request/response or send/receive_message.
2. Is there any difference between sending request vs sending response? If not, please refactor the code have common implementation of send and receive. Please avoid duplication of code.

Reply:
For sending request, the client/initiator uses `send_message_to_remote_host()` which calls `connectex()` for TCP and `WSASendTo` for UDP
For sending response, the server/responder uses `send_async_response()` which calls `WSASend` for TCP and `WSASendTo` to UDP.

So there is some overlap in case of UDP, but not for TCP. I will try to refactor this in this PR or file an issue to track this.